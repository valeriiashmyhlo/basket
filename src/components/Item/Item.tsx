import * as React from "react";
import { Card, Button } from "react-bootstrap";

interface ItemProps {
  name: string;
  price: number;
  onAddToCart: () => void;
}

const Item: React.FC<ItemProps> = ({ name, price, onAddToCart }) => {
  return (
    <Card style={{ width: "175px", margin: "10px" }}>
      <Card.Body>
        <Card.Title>{name}</Card.Title>
        <Card.Text>{price}</Card.Text>
        <Button onClick={onAddToCart}>Add to cart</Button>
      </Card.Body>
    </Card>
  );
};

export { Item };
