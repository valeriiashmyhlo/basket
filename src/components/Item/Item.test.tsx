import React from "react";
import { render, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import { Item } from "./Item";

describe(`<Item /> tests`, () => {
  it("should render Item with correct props", () => {
    const props = {
      id: '1',
      name: "Mask",
      price: 2.5,
    };
    const { container } = render(<Item onAddToCart={() => {}} {...props} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("should call onAddToCart", () => {
    const props = {
      id: '1',
      name: "Mask",
      price: 2.5,
    };
    const onClick = jest.fn();
    const { getByText } = render(<Item onAddToCart={onClick} {...props} />);
    fireEvent.click(getByText('Add to cart'));
    expect(onClick).toHaveBeenCalledTimes(1);
  });
});
