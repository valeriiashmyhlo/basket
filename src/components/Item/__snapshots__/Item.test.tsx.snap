// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item /> tests should render Item with correct props 1`] = `
<div
  class="card"
  style="width: 175px; margin: 10px;"
>
  <div
    class="card-body"
  >
    <div
      class="card-title h5"
    >
      Mask
    </div>
    <p
      class="card-text"
    >
      2.5
    </p>
    <button
      class="btn btn-primary"
      type="button"
    >
      Add to cart
    </button>
  </div>
</div>
`;
